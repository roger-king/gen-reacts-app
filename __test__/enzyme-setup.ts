import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({
    adapter: new Adapter()
});

export const shallowWithStore = (component, store) => {
    const context = {
        store
    };

    return shallow(component, {context});
};
