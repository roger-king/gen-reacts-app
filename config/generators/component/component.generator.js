var appModuleExists = require('./../../utils').appModuleExists;

module.exports = plop => {
    plop.setGenerator('component', {
        description: 'Create new shared component.',
        prompts: [
            {
                type: 'list',
                name: 'type',
                default: 'Stateless',
                message: 'Select the type of component?',
                choices: () => ['Stateless', 'React.PureComponent', 'React.Component'],
            },
            {
                type: 'input',
                name: 'name',
                message: 'Component name',
                validate: value => {
                    if (/.+/.test(value)) {
                        return appModuleExists(value) ? 'A component or container with this name already exists' : true;
                    }

                    return 'The name is required';
                },
            },
        ],
        actions: function(data) {
            var actions = [
                {
                    type: 'add',
                    path: './../../__test__/app/components/{{camelCase name}}.component.test.tsx',
                    templateFile: 'component/component.test.tsx.tpl',
                },
            ];

            switch (data.type) {
                case 'Stateless':
                    var component = {
                        type: 'add',
                        path: './../../src/app/components/{{camelCase name}}.tsx',
                        templateFile: 'component/sfcComponent.tsx',
                    };
                    actions = actions.concat(component);
                    break;
                case 'React.PureComponent':
                    var component = {
                        type: 'add',
                        path: './../../src/app/components/{{camelCase name}}.tsx',
                        templateFile: 'component/pureComponent.tsx',
                    };
                    actions = actions.concat(component);
                    break;
                case 'React.Component':
                    var component = {
                        type: 'add',
                        path: './../../src/app/components/{{camelCase name}}.tsx',
                        templateFile: 'component/component.tsx',
                    };
                    actions = actions.concat(component);
                    break;
                default:
                    break;
            }

            return actions;
        },
    });
};
