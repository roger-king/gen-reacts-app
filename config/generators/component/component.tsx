import * as React from 'react';

/**
 * {{pascalCase name}} Component
 */

export class ReactComponent extends React.Component<any, any> {
    constructor(props: any) {
        super(props);

        this.state = {};
    }

    render() {
        return <h1>React Component</h1>;
    }
}
