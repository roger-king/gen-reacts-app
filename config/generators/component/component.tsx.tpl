import * as React from 'react';

/**
 * {{pascalCase name}} Component
 */

interface I{{pascalCase name}}Props {
    name: string;
}

interface I{{pascalCase name}}State {}

export class {{pascalCase name}} extends React.Component <I{{pascalCase name}}Props, I{{pascalCase name}}State> {
    constructor(props: any) {
        super(props);

        this.state = {};
    }

    render() {
        return <h1>{this.props.name}</h1>;
    }
}
