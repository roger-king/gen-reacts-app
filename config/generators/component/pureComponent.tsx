import * as React from 'react';

/**
 * {{pascalCase name}} Pure Component
 */

interface I{{pascalCase name}}Props {
    name: string;
}

interface I{{pascalCase name}}State {}

export class {{pascalCase name}} extends React.PureComponent<I{{pascalCase name}}Props, I{{pascalCase name}}State> {
    constructor(props: any) {
        super(props);

        this.state = {};
    }

    render() {
        return <h1>{{pascalCase name}}</h1>;
    }
}
