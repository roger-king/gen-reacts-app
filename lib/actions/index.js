"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var createProject_1 = require("./createProject");
exports.createProject = createProject_1.createProject;
//# sourceMappingURL=index.js.map