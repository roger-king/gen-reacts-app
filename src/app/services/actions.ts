import Authentication from './authentication/authentication.action';

export default [].concat(Authentication);
