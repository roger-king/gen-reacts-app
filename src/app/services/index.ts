/**
 * Global imports of services.
 * This helps for cleaner imports through out the application.
 * i.e: auth services.
 */

// Global imports of services (do not remove - will break automation!)
