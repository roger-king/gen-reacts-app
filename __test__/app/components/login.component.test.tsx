import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import createMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Login, WrappedLogin } from './../../../src/app/components/login/login.component';
import { Login as LoginService } from './../../../src/app/services/actions';
import { shallowWithStore } from './../../enzyme-setup';

describe('<Login/>', function () {
    const initialState = {
        login: () => jest.fn,
        authentication: {
            loggedIn: false
        }
    };

    const mockStore = createMockStore();
    let store, container, wrapper;

    beforeEach(function () {
        store = mockStore(initialState);
        wrapper = shallowWithStore(<Login/>, store);
    });

    test('it renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });

    test('check props match with initialState', () => {
        expect(wrapper.find(WrappedLogin).prop('authentication.loggedIn')).toBeFalsy();
    });

    test('check action LOGIN is dispatching', () => {
        console.log(wrapper.state('username'));
    });
});
