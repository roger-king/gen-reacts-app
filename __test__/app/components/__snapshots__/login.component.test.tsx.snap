// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Snapshot Capture snapshot 1`] = `
<form
  className="login-container"
>
  <input
    className="arcing-input"
    name="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value=""
  />
  <input
    className="arcing-input"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <button
    className="arcing-btn"
    onClick={[Function]}
    type="button"
  >
     Login 
  </button>
  <p>
    
  </p>
</form>
`;
